server/applicationset/applicationset.proto

```diff
@@ -74,7 +74,7 @@ service ApplicationSetService {
 	// Generate generates
 	rpc Generate (ApplicationSetGenerateRequest) returns (ApplicationSetGenerateResponse) {
 		option (google.api.http) = {
-			post: "/api/v1/applicationsets"
+			put: "/api/v1/applicationsets"
```

Comment:
I think the convention is to use put for create, for other request that arent (crud), they should have a suffix.
```suggestion
			post: "/api/v1/applicationsets/generate"
```

Reply:
@agaudreault unless I'm mistaken, that will conflict with `GET /api/v1/applicationsets/{name}` assuming an applicationset is named `generate`. Maybe that's a tradeoff we're okay with taking, just making sure that's accepted